Updated Troubleshooting Function Guide w/Knative (#613)
vmware-samples/vcenter-event-broker-appliance: docs/kb/troubleshoot-functions.md

vmware-functions     default-broker-ingress-5c98bf68bc-2zpc6           1/1     Running     0          14h
vmware-functions     sockeye-65697bdfc4-cmfxc                          1/1     Running     0          14h
vmware-functions     sockeye-trigger-dispatcher-7f4dbd7f78-n589p       1/1     Running     0          14h
vmware-system        cadvisor-sk4j9                                    1/1     Running     0          14h
vmware-system        tinywww-dd88dc7db-dqnnc                           1/1     Running     0          14h
vmware-system        veba-rabbit-server-0                              1/1     Running     0          14h
vmware-system        veba-ui-54967b4bf4-lpjrn                          1/1     Running     0          14h
vmware-system        vmware-event-router-vcenter-6b76959df5-6mrb4      1/1     Running     3          14h
vmware-system        vmware-event-router-webhook-6b48cc5b8c-sjzx8      1/1     Running     0          14h
```

First, we want to see if the event router is capturing events and forwarding them on to a function.

Use this command to follow the live Event Router log.

```bash

> kubectl logs -n vmware-system vmware-event-router-vcenter-6b76959df5-6mrb4 -f

Reviewer: ```suggestion
kubectl logs -n vmware-system deploy/vmware-event-router-vcenter
```

Assuming this is the name of the deployment which avoids issue with copy and paste and different pod name

```

> For this sample troubleshooting, we have the sample [PowerCLI Tagging function](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powercli/kn-pcli-tag) running which will react to a VM powered on Event (`DrsVmPoweredOnEvent`). To see if the appliance is properly handling the event, we create a test VM and power it on.

Reviewer: ```suggestion
For this sample troubleshooting, we have the sample [PowerCLI Tagging function](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powercli/kn-pcli-tag) running which will react to a VM powered on Event (`DrsVmPoweredOnEvent`). To see if the appliance is properly handling the event, create a test VM and power it on before proceeding with the next steps.
```

Also text says "we have the tagging function running" but it's not deployed (seen in above all pods output). Might be confusing

When we look at the log output, we see various entries regarding DrsVmPoweredOnEvent, ending with the following:

```
2021-09-24T13:57:05.015Z	INFO	[KNATIVE]	knative/knative.go:181	sending event	{"eventID": "c6d61d55-8100-459e-a1e7-7a936bac6e43", "subject": "DrsVmPoweredOnEvent"}
2021-09-24T13:57:05.015Z	INFO	[KNATIVE]	knative/knative.go:193	successfully sent event	{"eventID": "c6d61d55-8100-459e-a1e7-7a936bac6e43"}
2021-09-24T13:57:06.017Z	INFO	[VCENTER]	vcenter/vcenter.go:343	invoking processor	{"eventID": "87af3e86-6516-4377-9a9b-86c4c9b00b05"}
```

This lets us know that the function was invoked. If we still don't see the expected result, we need to look at the function logs.

> Each Knative function will have its own pod running in the vmware-functions namespace. We can examine the logs with the following command.

Reviewer: Can you please add a note that the pod name has to be replaced by the actual name in the cluster? One comment is enough, i.e. not need to repeat for all log operations.

> Each Knative function will have its own pod running in the vmware-functions namespace. We can examine the logs with the following command.

Reviewer: ```suggestion
Each Knative function will have its own pod running in the vmware-functions namespace. If you have deployed the provided tagging function example from the VEBA examples, you can examine the logs with the following command.
```

```bash
kubectl logs -n vmware-functions kn-pcli-tag-00001-deployment-c845447d4-lnmrq user-container
```

> We don't need the --follow switch because we are just trying to look at recent logs, but --follow would work too.

Reviewer: For all flags used, can you please wrap them with ticks ``? (same with kubectl and other commands earlier)

Some other useful switches are `--since` and `--tail`.

This command will show you the last 5 minutes worth of logs.

```bash
kubectl logs -n vmware-functions kn-pcli-tag-00001-deployment-c845447d4-lnmrq user-container --since=5m
```

This command will show you the last 20 lines of logs.

```bash
kubectl logs -n vmware-functions kn-pcli-tag-00001-deployment-c845447d4-lnmrq user-container --tail=20
```

> Log output showing a succesful function invocation:

Reviewer: ```suggestion
Log output showing a successful function invocation:
```
